post.go

```diff
@@ -27,6 +27,23 @@ func (p *PostService) CreatePost(post *model.Post) error {
 	return nil
 }
 
+// DM sends a post as a direct message
+//
+// Minimum server version: 5.2
+func (p *PostService) DM(senderID, receiverID string, post *model.Post) (string, error) {
+	channel, appErr := p.api.GetDirectChannel(senderID, receiverID)
+	if appErr != nil {
+		return "", normalizeAppErr(appErr)
+	}
+	post.ChannelId = channel.Id
+	post.UserId = senderID
```

Comment:
While unorthodox, some plugin bots do post /into/ DMs, e.g. if a bot is invited to a two-person conversation to help with some automation. I'm not necessarily proposing that we support this explicitly here, but just wanted to note that we there can technically be three users involved in a DM, especially from a plugin's perspective.

Reply:
I wonder if it makes sense that being a completely different function, because you are no longer sending a DM, you are writing in a channel that happens to be a DM. If we were to put this into the DM function, most of the times we call DM would be something like:
```
DM(sender, sender, receiver, post)
```